### PR TITLE
Reworked Logger

### PR DESCRIPTION
**Pull request by Archisman-Mridha**
Leveraged JSON-based logging for Claudie.

### CODE REVIEW

File: internal/utils/log.go [hunk 1]
```diff
@@ -44,6 +45,14 @@ func InitLog(moduleName string) {
 	log.Logger = logger
 }
 
+// CreateLoggerWithProjectAndClusterName creates a new logger aware of the project-name and cluster-name
+// Returns the new logger
+func CreateLoggerWithProjectAndClusterName(projectName, clusterName string) zerolog.Logger {
```
Comment (MiroslavRepka) [inline internal/utils/log.go:50]: Let's check if the parameters are not `""`.

File: services/autoscaler-adapter/claudie_provider/claudie_node_groups.go [hunk 2]
```diff
@@ -23,7 +24,7 @@ func (c *ClaudieCloudProvider) NodeGroupTargetSize(_ context.Context, req *proto
 	defer c.lock.Unlock()
 	log.Info().Msgf("Got NodeGroupTargetSize request")
 	if ngc, ok := c.nodesCache[req.GetId()]; ok {
-		log.Debug().Msgf("Returning target size %d for nodepool %s", ngc.targetSize, req.GetId())
+		log.Debug().Str("nodepool", req.GetId()).Msgf("Returning target size %d for nodepool", ngc.targetSize)
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_node_groups.go:27]: Please, remove `for nodepool` part

File: services/autoscaler-adapter/claudie_provider/claudie_node_groups.go [hunk 3]
```diff
@@ -35,7 +36,7 @@ func (c *ClaudieCloudProvider) NodeGroupTargetSize(_ context.Context, req *proto
 func (c *ClaudieCloudProvider) NodeGroupIncreaseSize(_ context.Context, req *protos.NodeGroupIncreaseSizeRequest) (*protos.NodeGroupIncreaseSizeResponse, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	log.Info().Msgf("Got NodeGroupIncreaseSize request for nodepool %s by %d", req.GetId(), req.GetDelta())
+	log.Info().Str("nodepool", req.GetId()).Msgf("Got NodeGroupIncreaseSize request for nodepool by %d", req.GetDelta())
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_node_groups.go:39]: ditto

File: services/autoscaler-adapter/claudie_provider/claudie_node_groups.go [hunk 4]
```diff
@@ -61,7 +62,7 @@ func (c *ClaudieCloudProvider) NodeGroupIncreaseSize(_ context.Context, req *pro
 func (c *ClaudieCloudProvider) NodeGroupDeleteNodes(_ context.Context, req *protos.NodeGroupDeleteNodesRequest) (*protos.NodeGroupDeleteNodesResponse, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	log.Info().Msgf("Got NodeGroupDeleteNodes request for nodepool %s", req.GetId())
+	log.Info().Str("nodepool", req.GetId()).Msgf("Got NodeGroupDeleteNodes request for nodepool")
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_node_groups.go:65]: ditto

File: services/autoscaler-adapter/claudie_provider/claudie_provider.go [hunk 5]
```diff
@@ -209,7 +210,7 @@ func (c *ClaudieCloudProvider) Refresh(_ context.Context, req *protos.RefreshReq
 func (c *ClaudieCloudProvider) refresh() error {
 	log.Info().Msgf("Refreshing the state")
 	if cluster, err := getClaudieState(c.projectName, c.configCluster.ClusterInfo.Name); err != nil {
-		log.Error().Msgf("error while refreshing a state for the cluster %s : %v", c.configCluster.ClusterInfo.Name, err)
+		log.Err(err).Str("cluster", c.configCluster.ClusterInfo.Name).Msgf("error while refreshing a state for the cluster")
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_provider.go:213]: `error while refreshing a state for the cluster` -> `Error while refreshing a state of the cluster`

**Review by MiroslavRepka — COMMENTED** [review]
Good job @Archisman-Mridha!

File: services/ansibler/server/loadbalancers.go [hunk 6]
```diff
@@ -156,7 +156,7 @@ func teardownLoadBalancers(clusterName string, info *LBInfo, attached bool) (str
 	k8sDirectory := filepath.Join(baseDirectory, outputDirectory, fmt.Sprintf("%s-%s-lbs", clusterName, utils.CreateHash(utils.HashLength)))
 
 	if err := generateK8sBaseFiles(k8sDirectory, info); err != nil {
-		log.Err(err)
+		log.Error().Err(err)
```
Comment (MiroslavRepka) [inline services/ansibler/server/loadbalancers.go:159]: I am not sure this would send the message. I would expect some `Msg()` call at the end, i.e. like you did it here
```go
logger.Err(err).Msgf("Error encountered while installing node requirements")
```

File: services/ansibler/server/loadbalancers.go [hunk 7]
```diff
@@ -181,7 +181,7 @@ func setUpLoadbalancers(clusterName string, info *LBInfo) error {
 	directory := filepath.Join(baseDirectory, outputDirectory, fmt.Sprintf("%s-%s-lbs", clusterName, utils.CreateHash(utils.HashLength)))
 
 	if err := generateK8sBaseFiles(directory, info); err != nil {
-		log.Err(err)
+		log.Error().Err(err)
```
Comment (MiroslavRepka) [inline services/ansibler/server/loadbalancers.go:184]: ditto

File: services/autoscaler-adapter/claudie_provider/claudie_provider_test.go [hunk 8]
```diff
@@ -23,7 +23,7 @@ func TestRefresh(t *testing.T) {
 
 	defer func() {
 		if err := cc.Close(); err != nil {
-			log.Err(err).Msgf("Failed to close adapter connection")
+			log.Error().Err(err).Msgf("Failed to close adapter connection")
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_provider_test.go:26]:
```
Err starts a new message with error level with err as a field if not nil or with info level if err is nil.

You must call Msg on the returned event in order to send the event. 
```
I don't understand this change. From docs, the `Err()` already sets level of the message.

File: services/autoscaler-adapter/claudie_provider/claudie_provider.go [hunk 9]
```diff
@@ -210,7 +210,7 @@ func (c *ClaudieCloudProvider) Refresh(_ context.Context, req *protos.RefreshReq
 func (c *ClaudieCloudProvider) refresh() error {
 	log.Info().Msgf("Refreshing the state")
 	if cluster, err := getClaudieState(c.projectName, c.configCluster.ClusterInfo.Name); err != nil {
-		log.Err(err).Str("cluster", c.configCluster.ClusterInfo.Name).Msgf("error while refreshing a state for the cluster")
+		log.Error().Err(err).Str("cluster", c.configCluster.ClusterInfo.Name).Msgf("Error while refreshing a state of the cluster")
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_provider.go:213]: Ditto, don't understand this change

File: services/autoscaler-adapter/claudie_provider/claudie_node_groups.go [hunk 10]
```diff
@@ -24,7 +24,7 @@ func (c *ClaudieCloudProvider) NodeGroupTargetSize(_ context.Context, req *proto
 	defer c.lock.Unlock()
 	log.Info().Msgf("Got NodeGroupTargetSize request")
 	if ngc, ok := c.nodesCache[req.GetId()]; ok {
-		log.Debug().Str("nodepool", req.GetId()).Msgf("Returning target size %d for nodepool", ngc.targetSize)
+		log.Debug().Msgf("Returning target size %d for nodepool", ngc.targetSize)
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_node_groups.go:27]: Why did you remove the nodepool field here?

File: services/autoscaler-adapter/claudie_provider/claudie_node_groups.go [hunk 11]
```diff
@@ -36,7 +36,7 @@ func (c *ClaudieCloudProvider) NodeGroupTargetSize(_ context.Context, req *proto
 func (c *ClaudieCloudProvider) NodeGroupIncreaseSize(_ context.Context, req *protos.NodeGroupIncreaseSizeRequest) (*protos.NodeGroupIncreaseSizeResponse, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	log.Info().Str("nodepool", req.GetId()).Msgf("Got NodeGroupIncreaseSize request for nodepool by %d", req.GetDelta())
+	log.Info().Msgf("Got NodeGroupIncreaseSize request for nodepool by %d", req.GetDelta())
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_node_groups.go:39]: ditto

File: services/autoscaler-adapter/claudie_provider/claudie_node_groups.go [hunk 12]
```diff
@@ -62,7 +62,7 @@ func (c *ClaudieCloudProvider) NodeGroupIncreaseSize(_ context.Context, req *pro
 func (c *ClaudieCloudProvider) NodeGroupDeleteNodes(_ context.Context, req *protos.NodeGroupDeleteNodesRequest) (*protos.NodeGroupDeleteNodesResponse, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	log.Info().Str("nodepool", req.GetId()).Msgf("Got NodeGroupDeleteNodes request for nodepool")
+	log.Info().Msgf("Got NodeGroupDeleteNodes request for nodepool")
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_node_groups.go:65]: ditto

File: services/builder/configProcessor.go [hunk 13]
```diff
@@ -60,38 +60,38 @@ func configProcessor(c pb.ContextBoxServiceClient, wg *sync.WaitGroup) error {
 		if config.DsChecksum == nil && config.CsChecksum != nil {
 			if err := destroyConfig(config, clusterView, c); err != nil {
 				// Save error to DB.
-				log.Err(err).Str("project", config.Name).Msgf("Error while destroying")
+				log.Error().Err(err).Str("project", config.Name).Msgf("Error while destroying")
```
Comment (MiroslavRepka) [inline services/builder/configProcessor.go:63]: ditto, why add `Error()`?

File: services/builder/configProcessor.go [hunk 14]
```diff
@@ -60,38 +60,38 @@ func configProcessor(c pb.ContextBoxServiceClient, wg *sync.WaitGroup) error {
...
 				if err := saveConfigWithWorkflowError(config, c, clusterView); err != nil {
-					log.Err(err).Str("project", config.Name).Msgf("Failed to save error message")
+					log.Error().Err(err).Str("project", config.Name).Msgf("Failed to save error message")
```
Comment (MiroslavRepka) [inline services/builder/configProcessor.go:65]: ditto, why add `Error()`?

File: services/builder/configProcessor.go [hunk 15]
```diff
@@ -148,46 +148,46 @@ func configProcessor(c pb.ContextBoxServiceClient, wg *sync.WaitGroup) error {
...
 			}
 
 			// Propagate the changes made to the cluster back to the View.
 			clusterView.UpdateFromBuild(ctx)
-			_logger.Info().Msgf("Finished building cluster")
+			logger.Info().Msgf("Finished building cluster")
 			return nil
 		}); err != nil {
-			log.Err(err).Str("project", config.Name).Msgf("Error encountered while processing config")
+			log.Error().Err(err).Str("project", config.Name).Msgf("Error encountered while processing config")
```
Comment (MiroslavRepka) [inline services/builder/configProcessor.go:168]: ditto, why add `Error()`?

File: services/builder/configProcessor.go [hunk 16]
```diff
@@ -148,46 +148,46 @@ func configProcessor(c pb.ContextBoxServiceClient, wg *sync.WaitGroup) error {
...
 			if err := saveConfigWithWorkflowError(config, c, clusterView); err != nil {
-				log.Err(err).Msgf("Failed to save error message")
+				log.Error().Err(err).Msgf("Failed to save error message")
```
Comment (MiroslavRepka) [inline services/builder/configProcessor.go:170]: ditto, why add `Error()`?

File: services/context-box/server/claudieDB/claudieMongo.go [hunk 17]
```diff
@@ -437,7 +437,7 @@ func (c *ClaudieMongo) getAllFromDB() ([]*configItem, error) {
 	defer func() {
 		err := cur.Close(context.Background())
 		if err != nil {
-			log.Err(err).Msgf("Failed to close MongoDB cursor")
+			log.Error().Err(err).Msgf("Failed to close MongoDB cursor")
```
Comment (MiroslavRepka) [inline services/context-box/server/claudieDB/claudieMongo.go:440]: ditto, why add `Error()`?

File: services/frontend/adapters/inbound/k8s_sidecar_notifications_receiver.adapter.go [hunk 18]
```diff
@@ -123,7 +123,7 @@ func (k *K8sSidecarNotificationsReceiver) processManifestFilesHandler(responseWr
 		defer k.waitGroup.Done()
 
 		if err := k.usecases.ProcessManifestFiles(k.manifestDir); err != nil {
-			log.Err(err).Msgf("Failed processing manifest files from directory %s", k.manifestDir)
+			log.Error().Err(err).Msgf("Failed processing manifest files from directory %s", k.manifestDir)
```
Comment (MiroslavRepka) [inline services/frontend/adapters/inbound/k8s_sidecar_notifications_receiver.adapter.go:126]: ditto, why add `Error()`?

File: services/frontend/domain/usecases/process_manifest_files.go [hunk 19]
```diff
@@ -92,7 +92,7 @@ func (u *Usecases) ProcessManifestFiles(manifestDir string) error {
 		configs, isConfigRemoved = removeConfig(configs, manifestProcessingResult.manifestFilepath)
 		// Check for the error first, before referencing any variables.
 		if manifestProcessingResult.processingError != nil {
-			log.Err(manifestProcessingResult.processingError).
+			log.Error().Err(manifestProcessingResult.processingError).
```
Comment (MiroslavRepka) [inline services/frontend/domain/usecases/process_manifest_files.go:95]: ditto, why add `Error()`?

File: services/frontend/domain/usecases/process_manifest_files.go [hunk 20]
```diff
@@ -107,7 +107,7 @@ func (u *Usecases) ProcessManifestFiles(manifestDir string) error {
 
 		err = u.ContextBox.SaveConfig(config)
 		if err != nil {
-			log.Err(err).Str("project", manifestName).Msgf("Failed to save config")
+			log.Error().Err(err).Str("project", manifestName).Msgf("Failed to save config")
```
Comment (MiroslavRepka) [inline services/frontend/domain/usecases/process_manifest_files.go:110]: ditto, why add `Error()`?

File: services/frontend/domain/usecases/process_manifest_files.go [hunk 21]
```diff
@@ -144,7 +144,7 @@ func (u *Usecases) ProcessManifestFiles(manifestDir string) error {
 
 			err := u.ContextBox.DeleteConfig(config.Id)
 			if err != nil {
-				log.Err(err).
+				log.Error().Err(err).
```
Comment (MiroslavRepka) [inline services/frontend/domain/usecases/process_manifest_files.go:147]: ditto, why add `Error()`?

File: services/frontend/domain/usecases/watch_configs.go [hunk 22]
```diff
@@ -17,7 +17,7 @@ func (u *Usecases) WatchConfigs() {
 
 	configs, err := u.ContextBox.GetAllConfigs()
 	if err != nil {
-		log.Err(err).Msgf("Failed to retrieve configs from context-box")
+		log.Error().Err(err).Msgf("Failed to retrieve configs from context-box")
```
Comment (MiroslavRepka) [inline services/frontend/domain/usecases/watch_configs.go:20]: ditto, why add `Error()`?

File: services/frontend/domain/usecases/watch_configs.go [hunk 23]
```diff
@@ -36,7 +36,7 @@ func (u *Usecases) WatchConfigs() {
 			{
 				configs, err = u.ContextBox.GetAllConfigs()
 				if err != nil {
-					log.Err(err).Msgf("Failed to retrieve configs from context-box")
+					log.Error().Err(err).Msgf("Failed to retrieve configs from context-box")
```
Comment (MiroslavRepka) [inline services/frontend/domain/usecases/watch_configs.go:39]: ditto, why add `Error()`?

**Review by MiroslavRepka — CHANGES_REQUESTED** [review]
Hi @Archisman-Mridha, thanks for the changes, but I think you overcomplicated it a bit by making a certain change.

There are many instances, where you call 
```go
log.Error().Err(err).Msgf("some message")
```

Which just does not make sense, as previously you had 
```go
log.Err(err).Msgf("some message")
```
which was just fine. I pasted the doc snippet why it's not necessary to call `Error()`. 

Bear in mind there are many instances of this where I did not leave a comment, please double-check the code yourself.

File: services/ansibler/server/loadbalancers.go [hunk 24]
```diff
@@ -156,7 +156,7 @@ func teardownLoadBalancers(clusterName string, info *LBInfo, attached bool) (str
 	k8sDirectory := filepath.Join(baseDirectory, outputDirectory, fmt.Sprintf("%s-%s-lbs", clusterName, utils.CreateHash(utils.HashLength)))
 
 	if err := generateK8sBaseFiles(k8sDirectory, info); err != nil {
-		log.Error().Err(err)
+		log.Err(err).Msg("")
```
Comment (MiroslavRepka) [inline services/ansibler/server/loadbalancers.go:159]: Can we add some description to this error?

File: services/ansibler/server/loadbalancers.go [hunk 25]
```diff
@@ -156,7 +156,7 @@ func teardownLoadBalancers(clusterName string, info *LBInfo, attached bool) (str
 	k8sDirectory := filepath.Join(baseDirectory, outputDirectory, fmt.Sprintf("%s-%s-lbs", clusterName, utils.CreateHash(utils.HashLength)))
 
 	if err := generateK8sBaseFiles(k8sDirectory, info); err != nil {
-		log.Err(err).Msg("")
+		log.Err(err).Str("lb-cluster", info.ClusterID).Msg("Error base loadbalancer files")
```
Comment (MiroslavRepka) [inline services/ansibler/server/loadbalancers.go:159]: `Error base loadbalancer files` -> `Error encountered while generating base files`

File: services/ansibler/server/loadbalancers.go [hunk 26]
```diff
@@ -181,7 +181,7 @@ func setUpLoadbalancers(clusterName string, info *LBInfo) error {
 	directory := filepath.Join(baseDirectory, outputDirectory, fmt.Sprintf("%s-%s-lbs", clusterName, utils.CreateHash(utils.HashLength)))
 
 	if err := generateK8sBaseFiles(directory, info); err != nil {
-		log.Err(err).Msg("")
+		log.Err(err).Str("lb-cluster", info.ClusterID).Msg("Error base loadbalancer files")
```
Comment (MiroslavRepka) [inline services/ansibler/server/loadbalancers.go:184]: Ditto

**Review by Despire — APPROVED** [review]
LGTM 👍 I like the simplification of the logs

**Review by MiroslavRepka — APPROVED** [review]
